checkov/secrets/runner.py

```diff
@@ -231,12 +234,8 @@ def run(
                     logging.info(
                         f"Removing secret due to UUID filtering: {hashlib.sha256(secret.secret_value.encode('utf-8')).hexdigest()}")
                     continue
-                if secret_key in secrets_duplication:
-                    logging.debug(
-                        f'Secret was filtered - secrets_duplication. line_number {secret.line_number}, check_id {check_id}')
-                    continue
-                else:
-                    secrets_duplication[secret_key] = True
+                if secret_key in secret_records.keys() and secret_records[secret_key].check_id in ENTROPY_CHECK_IDS and check_id not in ENTROPY_CHECK_IDS:
+                    secret_records.pop(secret_key)
```

Comment:
don't you need to also add it back in the end?

Reply:
Nope, we just removed the existing and the new will add in this iteration.